### PR TITLE
refactor: restructure documentation

### PR DESCRIPTION
**Pull request by EmanuelCampos**

### CODE REVIEW

**Review by marcusschiesser — APPROVED** [review]

**Comment by marcusschiesser** [comment]
@EmanuelCampos LGTM, but can't merge

**Comment by EmanuelCampos** [comment]
> @EmanuelCampos LGTM, but can't merge

due to conflicts? fixed they now

File: apps/docs/docs/examples/chat_engine.md
Comment (himself65) [inline apps/docs/docs/examples/chat_engine.md:1]: I'm a little worry about the maintenance of the example code. Could we read raw text from `<rootDir>/examples` and render them?

**Review by himself65 — CHANGES_REQUESTED** [review]
I will do some change on code example

**Review by himself65 — APPROVED** [review]

**Comment by marcusschiesser** [comment]
@EmanuelCampos thanks!

**Pull request merged**